frontend/test/metabase/scenarios/sharing/reproductions/25473-dashboard-text-filter-asking-for-number.cy.spec.js

```diff
@@ -0,0 +1,117 @@
+import {
+  restore,
+  visitEmbeddedPage,
+  filterWidget,
+  visitPublicDashboard,
+} from "__support__/e2e/helpers";
+
+import { SAMPLE_DATABASE } from "__support__/e2e/cypress_sample_database";
+
+const { REVIEWS, REVIEWS_ID } = SAMPLE_DATABASE;
+
+const ccName = "CC Reviewer";
+
+const dashboardFilter = {
+  name: "Text ends with",
+  slug: "text_ends_with",
+  id: "3a8ecdbd",
+  type: "string/ends-with",
+  sectionId: "string",
+};
+
+const questionDetails = {
+  name: "25473",
+  query: {
+    "source-table": REVIEWS_ID,
+    expressions: { [ccName]: ["field", REVIEWS.REVIEWER, null] },
+    limit: 10,
+    // Let's show only a few columns to make it easier to focus on the UI
+    fields: [
+      ["field", REVIEWS.REVIEWER, null],
+      ["field", REVIEWS.RATING, null],
+      ["field", REVIEWS.CREATED_AT, null],
+      ["expression", ccName, null],
+    ],
+  },
+};
+
+const dashboardDetails = {
+  name: "25473D",
+  parameters: [dashboardFilter],
+};
+
+describe("issue 25473", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+
+    cy.createQuestionAndDashboard({ questionDetails, dashboardDetails }).then(
+      ({ body: { id, card_id, dashboard_id } }) => {
+        cy.request("PUT", `/api/dashboard/${dashboard_id}/cards`, {
+          cards: [
+            {
+              id,
+              card_id,
+              row: 0,
+              col: 0,
+              size_x: 12,
+              size_y: 8,
+              series: [],
+              visualization_settings: {},
+              parameter_mappings: [
+                {
+                  parameter_id: dashboardFilter.id,
+                  card_id,
+                  target: ["dimension", ["expression", ccName, null]],
+                },
+              ],
+            },
+          ],
+        });
+
+        cy.wrap(dashboard_id).as("dashboardId");
+      },
+    );
+  });
+
+  it("public sharing: dashboard text filter on a custom column should accept text input (metabase#25473-1)", () => {
+    cy.get("@dashboardId").then(id => {
+      visitPublicDashboard(id);
```

Comment:
This got me until I went to its definition. Maybe what if we rename it to be more precise `makeDashboardPublicAndVisit` or something along that line.

This gave me the impression that we only visit the **already** publicly shared dashboard. But this is minor.